collapsible-selector.js

```diff
@@ -0,0 +1,403 @@
+'use strict';
+
+(function() {
+  Polymer({
+    is: 'collapsible-selector',
+
+    properties: {
+      /*********************/
+      /* Public properties */
+      /*********************/
+
+      /**
+       * The text to display on the dropdown menu button.
+       */
+      dropdownButtonText: {
+        type: String,
+        value: 'More...'
+      },
+
+      /**
+       * The list of items that the collapsible-selector should display.
+       *
+       * Each element in the list should be in the format:
+       *
+       * {
+       *   display: 'This is an item',
+       *   value: 'itemId'
+       * }
+       */
+      items: {
+        type: Array,
+        value: [],
+        observer: '_itemsChanged'
+      },
+
+      /**
+       * Name of an event to listen to which should trigger a recalculation of visible items.
+       *
+       * Ideally this would be triggered when the width of the element containing the collapsible-selector
+       * changes.
+       *
+       * The selector will always listen for window.onresize events.
+       */
+      resizeEventName: {
+        type: String,
+        observer: '_resizeEventNameChanged'
+      },
+
+      /**
+       * Wait in milliseconds to debounce the resize event handler.
+       */
+      resizeWaitTime: {
```

Comment:
What is the basis for choosing 20? Is it better to make this number as small as possible to make it appear as responsive as the machine will allow, or to make this as high as possible so that it consumes the least amount of resources.